packages/android-client/FishjamClient/src/main/java/com/fishjamcloud/client/CommandsQueue.kt

```diff
@@ -1,28 +1,31 @@
 package com.fishjamcloud.client
 
-import kotlinx.coroutines.CancellationException
-import kotlinx.coroutines.Job
+import kotlin.coroutines.resume
+import kotlin.coroutines.suspendCoroutine
 
 internal class CommandsQueue {
   private var commandsQueue: ArrayDeque<Command> = ArrayDeque()
   var clientState: ClientState = ClientState.CREATED
 
-  fun addCommand(command: Command): Job {
+  suspend fun executeCommand(command: Command) {
     commandsQueue.add(command)
-    if (commandsQueue.size == 1) {
-      command.execute()
+    suspendCoroutine { cont ->
```

Comment:
why this must be a suspendCorutine?

Reply:
so we can await for finishing the command